cmd/juno/main.go

```diff
@@ -188,7 +190,12 @@ func setFlagValue(cmd *cobra.Command, v *viper.Viper, f *pflag.Flag) {
 
 // juno is the main entrypoint for the Juno node.
 func juno(cfg *config.Juno) {
-	fmt.Printf("using config: %+v\n\n", cfg)
+	configYaml, err := yaml.Marshal(&cfg)
+	if err != nil {
+		log.Fatal("Failed to marshal config: ", err)
```

Comment:
Although we are importing Logger in this file, we don't activate it until after printing the config. See [this](https://github.com/NethermindEth/juno/blob/main/cmd/juno/main.go#L194)

Reply:
I think the config is being "pretty printed" much like the banner so it should probably use `fmt.Printf` instead of the logger, @adriantpaez.

@jelilat, this is where manually implementing the `fmt.Stringer` interface would help because we wouldn't have to handle an error before the logger is initialised. It might be slightly more tedious than marshalling YAML though.

It would also make it easier to bold or colour the keys using [fatih/color](https://github.com/fatih/color) but that would probably require a second method say `Color` so that we can branch in `main` depending on whether the user had the colour parameter set.

Reply:
@tshakalekholoane So, I manually implemented the `fmt.Stringer` and here's how it currently looks 
<img width="752" alt="Screenshot 2022-08-17 at 02 10 45" src="https://user-images.githubusercontent.com/23613565/185012983-985c6ce4-b57a-4ae5-aec5-865d08ca41b0.png">

I need recommendations on the colour and better formatting options so I can make changes before I push

Reply:
Option 2

<img width="830" alt="Screenshot 2022-08-17 at 02 30 56" src="https://user-images.githubusercontent.com/23613565/185014889-07ecd1c1-aad2-4cc3-a92e-6328908217f5.png">

Reply:
This looks great @jelilat! I think for visual consistency, we should do either the table or list the key-value pairs in "yaml" format but not both.

A better idea might be to turn this PR into a draft and then wait to see the changes coming from what @IronGauntlets is working on around the configuration. Then if we still have enough common fields we can do a table but if not then go with a simple yaml format with the option of having the key coloured.

What do you think about that?

Reply:
Alright. That works

Reply:
Looks really good!

Reply:
I will mark these requests as resolved and we wait then for the merge of @IronGauntlets changes on the config